codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -548,6 +585,37 @@ class KotlinJmespathExpressionVisitor(
         return notNull
     }
 
+    private fun mapFunctionLogic(
```

Comment:
I'd call this `applyFunction`, since `mapFunctionLogic` could be confused for the actual `map` function, and I think the `Logic` suffix is unnecessary

Reply:
Would it make sense to make this an extension function of `JmespathExpression`? Then you can use it like this

```kt
private fun JmespathExpression.applyFunction(
   ...
) { ... }

expression.applyFunction(arg, "sorted", "sortedBy")
```